### PR TITLE
feat: support custom client id (as appName) in metadata

### PR DESCRIPTION
**Pull request by dominique-pfister**

### CODE REVIEW

File: src/OneDriveAuth.js [hunk 1]
```diff
@@ -110,16 +125,25 @@ export class OneDriveAuth {
...
-      const {
-        log,
-        cachePlugin,
-      } = this;
+      const { log, cachePlugin } = this;
+
+      const metadata = await cachePlugin.getPluginMetadata();
+      if (metadata?.useClientCredentials) {
+        this.acquireMethod = AcquireMethod.BY_CLIENT_CREDENTIAL;
+      }
```
Comment (dominique-pfister) [inline src/OneDriveAuth.js:140]: overwrite `acquireMethod` here instead of later

File: test/OneDriveAuth.test.js [hunk 2]
```diff
@@ -190,6 +190,80 @@ describe('OneDriveAuth Tests', () => {
...
+    const cachePlugin = new MemCachePlugin({
+      key: 'default',
+      caches,
+      type: 'onedrive',
+    });
+    caches.set('default', {
+      metadata: {
+        useClientCredentials: true,
+        appName: 'FRANKLIN_CONNECTOR',
+      },
```
Comment (dominique-pfister) [inline test/OneDriveAuth.test.js:224]: `metadata` with new `appName`

File: test/OneDriveAuth.test.js [hunk 3]
```diff
@@ -302,6 +376,7 @@ describe('OneDriveAuth Tests', () => {
           beforeCacheAccess: () => {
             baseRefreshed = true;
           },
+          getPluginMetadata: () => {},
```
Comment (dominique-pfister) [inline test/OneDriveAuth.test.js:379]: the mock base used here needs that method as well (or we need to guard the call to `base.getPluginMetadata` in `MemCachePlugin`)

File: src/OneDriveAuth.js [hunk 4]
```diff
@@ -46,11 +46,24 @@ export const AcquireMethod = {
...
 const globalTenantCache = new Map();
 
+/**
+ * Return client id and secret stored in the process, matching an application name given.
+ * @param {String} appName application name, e.g. `HELIX_SERVICE`, or undefined
+ * @returns {Object} containing `clientId` and `clientSecret` or an empty object
+ */
+function getClientIdSecret(appName) {
+  if (appName) {
+    const clientId = process.env[`AZURE_${appName}_CLIENT_ID`];
```
Comment (tripodsan) [inline src/OneDriveAuth.js:56]: maybe `toUpperCase` ?

File: src/OneDriveAuth.js [hunk 5]
```diff
@@ -46,11 +46,24 @@ export const AcquireMethod = {
  */
 const globalTenantCache = new Map();
 
+/**
+ * Return client id and secret stored in the process, matching an application name given.
+ * @param {String} appName application name, e.g. `HELIX_SERVICE`, or undefined
+ * @returns {Object} containing `clientId` and `clientSecret` or an empty object
+ */
+function getClientIdSecret(appName) {
```
Comment (tripodsan) [inline src/OneDriveAuth.js:54]:
```suggestion
function getClientIdAndSecret(appName) {
```

**Review by tripodsan — APPROVED** [review]

File: src/OneDriveAuth.js [hunk 6]
```diff
@@ -110,16 +125,25 @@ export class OneDriveAuth {
...
     if (!this._app) {
-      const {
-        log,
-        cachePlugin,
-      } = this;
+      const { log, cachePlugin } = this;
+
+      const metadata = await cachePlugin.getPluginMetadata();
+      if (metadata?.useClientCredentials) {
+        this.pluginUseClientCredentials = true;
```
Comment (dominique-pfister) [inline src/OneDriveAuth.js:139]: use separate field, do not overwrite `acquireMethod` passed in constructor

**Review by tripodsan — APPROVED** [review]

**Pull request merged**